tests/pjit_test.py

```diff
@@ -1295,14 +1295,30 @@ def f(x):
         pjit_f(jnp.array([1, 2, 3]))
 
   def test_pretty_print(self):
-    f = pjit(lambda x: x)
+    f = pjit(lambda x: x**2)
+    g = pjit(lambda x: f(x) + f(x))
     x = jnp.array([4.2], dtype=jnp.float32)
-    jaxpr = jax.make_jaxpr(f)(x)
+    jaxpr = jax.make_jaxpr(g)(x)
     self.assertEqual(
         jaxpr.pretty_print(),
         textwrap.dedent("""
             { lambda ; a:f32[1]. let
-                b:f32[1] = pjit[name=<lambda> jaxpr={ lambda ; c:f32[1]. let  in (c,) }] a
+                b:f32[1] = pjit[
```

Comment:
This is an improvement, but my original suggestion was to add a new top-level `let` declaration for each function that appears twice. Would that be possible?

Reply:
+1 to printing the seen jaxprs at the top of the top-level